ls1intum/Artemis
# Allow multiple submissions in modeling exercises

### Checklist

### Motivation and Context


We want to allow students to submit exercise until the exercise deadline.
The are following criteria for allowing to submit an exercise are the following:
    - There is no due date
    - The due date is before the current date
    - The participation initialization date is after the due date
    - The result is not yet set (avoid infinite submission for late and without due date submissions)

### Description

The Save button is gone, but the student can submit as much as needed and last submission before the exercise due date will be reviewed. If student started participating in the exercise after the due date the submit button will be called `Submit(deadline missed)`. Tooltips added for both buttons. 

### Steps for Testing


1. Log in to Artemis
2. Navigate to Overview
3. Select Course with Modelling Exercise
4. Start new Modeling Exercise
5. Submit exercise multiple time
6. Wait for exercise deadline and check that Submit button is disabled now

### Screenshots



![Peek 2019-11-09 15-27](https://user-images.githubusercontent.com/17262376/68530121-9a7fad80-0305-11ea-99fc-a61dc999e1fc.gif)
